spec/test/stored_queries/test_fulltext_search_flags.py

```diff
@@ -0,0 +1,287 @@
+"""
+Tests the boolean flags on NCBI Taxon nodes when used in conjunction with full text search, e.g.
+in the queries:
+- taxonomy_search_species_strain
+- taxonomy_search_species_strain_no_sort
+
+Note these tests omit several fields from the records that are not relevant to the queries.
+"""
+
+import os
+import yaml
+from arango import ArangoClient
+from frozendict import frozendict  # type: ignore[attr-defined]
+from pathlib import Path
+from pytest import fixture
+from relation_engine_server.utils.config import get_config
+from relation_engine_client import REClient
+
+_QUERY_SORT = "taxonomy_search_species_strain"
+_QUERY_NO_SORT = "taxonomy_search_species_strain_no_sort"
+
+
+@fixture(scope="module")
+def conf():
+    config = get_config()
+    config["re_api_url"] = os.environ.get("RE_API_URL", "http://localhost:5000")
+
+    yield config
+
+
+@fixture(scope="module")
+def spec(conf):
+    # There should probably be some sort of abstraction here so this isn't hard coded, but AFAICT
+    # it doesn't exist in the code base at this point. Add later maybe. At least the tests
+    # should fail if the file is renamed / altered
+    schemapath = Path(conf["spec_paths"]["collections"]) / "ncbi" / "ncbi_taxon.yaml"
+    with open(schemapath) as fd:
+        spec = yaml.safe_load(fd)
+    yield spec
+
+
+@fixture(scope="module")
+def db(conf, spec):
+    # This is set to the "_system" db by the get_config() method defaults, which are used to start
+    # up the RE API in the docker tests...
+    arango = ArangoClient(conf["db_url"])
+    db = arango.db(conf["db_name"])
+
+    yield db
+
+    # clean up for tests in other modules that might be using the same db / collection
+    _clear_collection(db, spec["name"])
+
+
+@fixture(scope="module")
+def re_client(conf):
+    re_c = REClient(conf["re_api_url"], "non_admin_token")
+    yield re_c
+
+
+@fixture()
+def col(db, spec):
+    yield _clear_collection(db, spec["name"])
```

Comment:
Curious. I heard it was un-pythonic to return "complex expressions"

Reply:
So 
```
ret = _clear_collection(db, spec["name"])
return ret
```
?

Seems like an extra line without any gain in readability to me. I suppose it depends what "complex expressions" means - to me a function call isn't complex...

Reply:
I think the argument is that breaking up any complex expression (realpython's example was mathematical) with temporary variables makes it more readable/robust by computing subexpressions and giving them temporary variables.

Returning a function call may not qualify as a "complex expression", and I've certainly done it. But, I would argue it sort of has the same effect if the function name does not make it clear what is being returned

So, like
```
@fixture()
def coll(db, spec):
    db = _clear_collection(db, spec["name"])
    yield db


def _clear collection(db: arango.database.StandardDatabase, coll_name: str) -> arango.databse.StandardDatabase:
    collection = db.collection(coll_name)
    collection.truncate()

    return collection
```

Reply:
> I think the argument is that breaking up any complex expression (realpython's example was mathematical) with temporary variables makes it more readable/robust by computing subexpressions and giving them temporary variables.

Completely agree, and I think this rule applies everywhere, not just in return statements. However, I don't see a way to break up this function call into smaller parts.

> the function name does not make it clear what is being returned

You're right, I've renamed the function to make it more clear what's going on. My name is slightly different than yours to make it clear that the collection might already exist.

Reply:
I was just relaying the original argument I had seen for not returning complex expressions, which of course don't apply precisely to this situation.

Reply:
yeah, the equivalent of what's going on here outside of the context of a return would be something like

```
tempvar1 = a + b
tempvar2 = c + d
tempvar3 = tempvar1 + tempvar2
finalvar = tempvar3
```

Here `tempvar3` doesn't help the readability of the code at all and is just taking up space

Reply:
Just to play devil's advocate, I don't see any point in this kind of thing:

```
def coll(db, spec):
    coll = _get_cleared_collection(db, spec["name"])
    yield coll
```

This is a needless extra variable and IMO it decreases readability because the function is now on two lines when it could easily be expressed in a single line. There's more of an argument if you wanted to add in some debugging stuff whilst during development, e.g.

```
def coll(db, spec):
    coll = _get_cleared_collection(db, spec["name"])
    # during development, you might want to do something like:
    # print("coll = " + coll)
    new_coll = coll + "_new"
    return new_coll
```

Otherwise I would always advocate for getting rid of unnecessary variables.

Flake8 plugin for removing unnecessary variables before return statements: https://pypi.org/project/flake8-return/

Reply:
No disagreement here

Reply:
At this point in my journey I neither agree nor disagree. But when doing something like `return some_func()`, I could only recommend that if `some_func()` clearly indicates what is being returned, usually by function name and function signature documentation.

Reply:
Agreed, and I think it does that now after you pointed that out - or are there more changes needed IYO?

Reply:
hah, I didn't realize I was replying to AJ and not you earlier until just now